search/search_repository.go

```diff
@@ -265,9 +274,14 @@ func (r *GormSearchRepository) search(ctx context.Context, sqlSearchQueryParamet
 		}
 		db = db.Limit(*limit)
 	}
+	if len(workItemTypes) > 0 {
+		query := fmt.Sprintf("%[1]s.type in (select distinct w1.name from %[2]s w1 join %[2]s w2 on w1.path like (w2.path || '%%') where w2.name in (?))", models.WorkItem{}.TableName(), models.WorkItemType{}.TableName())
```

Comment:
Can you please elaborate on this query a bit? Also, please rename table aliases from `w1` to `wit1` and from `w2` to `wit2`. This should make things easier to understand. For clarity I've written it like so:

```sql
work_items.type in (
   SELECT DISTINCT wit1.name
   FROM work_item_types wit1
   JOIN work_item_types wit2
      ON wit1.path
      LIKE (wit2.path || '%%')
   WHERE wit2.name in (?)
)
```

I'd suggest to write it like this in Go:

```go
query := `%[1]s.type in (
   SELECT DISTINCT wit1.name
   FROM %[2]s wit1
   JOIN %[2]s wit2
      ON wit1.path
      LIKE (wit2.path || '%%')
   WHERE wit2.name in (?)
)`
```

That makes it much easier to understand IMHO.